projects/vdk-plugins/vdk-notebook/src/vdk/plugin/notebook/cell.py

```diff
@@ -1,35 +1,42 @@
 # Copyright 2021 VMware, Inc.
 # SPDX-License-Identifier: Apache-2.0
+from dataclasses import dataclass
 
 
+@dataclass
 class Cell:
     """
-    Helper class that retrieves data from Jupyter cells
-    Only the data essential for running a VDK data job is saved
-    Other data is ignored
+    A class that represents Jupyter code cell
     """
 
     def __init__(self, jupyter_cell):
         self.tags = jupyter_cell["metadata"].get("tags", {})
         self.source = "".join(jupyter_cell["source"])
 
-    def is_vdk_cell(self):
-        return "vdk" in self.tags
 
-    def is_sql_cell(self):
-        return self.source.startswith("%sql")
+class CellFuncFactory:
+    @staticmethod
+    def is_vdk_cell(cell: Cell):
+        return "vdk" in cell.tags
 
-    def is_vdk_run_cell(self):
-        return "def run(" in self.source
+    @staticmethod
+    def is_sql_cell(cell: Cell):
+        return cell.source.startswith("%sql")
 
-    def add_code(self, cells):
+    @staticmethod
+    def is_vdk_run_cell(cell: Cell):
+        return "def run(" in cell.source
+
```

Comment:
You should very very very rarely rely on "custom" parsing logic (that is not following some standard (e.g json)) ever. OK? Never do that again, please. 
 It's so easy to break and have corner cases you cannot even think of currently. This will not work long term. Period. This line will have bugs 99.999% sure. 

Instead, you should use the python parser to parse the file and see if there's a method called run there. 

While We need to fix that. I missed that in previous review, and this PR is about other things. So create a story and will fix that later.